### ISSUE TITLE
Unused, undeclared array not flagged

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1909]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online and 0.7.0 on my computer
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

why_does_assignment_result_in_2034=(1)
but_adding_to_does_not+=(1)

echo some dummy string
```

#### Here's what shellcheck currently says:
```
Line 3	SC2034: why_does_assignment_result_in_2034 appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:
```
Line 3	SC2034: why_does_assignment_result_in_2034 appears unused. Verify use (or export if used externally).
Line 4	SC2034: but_adding_to_does_not appears unused. Verify use (or export if used externally).
```
#### Comments:
This caused me an bit of head scratching this morning:

```
...
declare -a _profile_env_init_list
## Notice the missing lead '_'?
profile_env_init_list+=('util.env' 'cd.env' 'perl.env' 'git.env')
profile_env_init_list+=('ansi-colors.env' 'xtermfuncs' 'python.env')
if [ "$OS_ID" = 'WSL' ]
then
    _profile_env_init_list+=('wsl.env')
fi

declare _profile_env _profile_envvar _profile_envvar_val
## So that this loop never ran and I had no env setups for my login
for _profile_env in "${_profile_env_init_list[@]}"
do
    _profile_envvar="$(echo "${_profile_env^^}" | tr -d '.' | tr '-' '_')_LOADED"
    _profile_envvar_val="$(eval echo "\$$_profile_envvar")"
    if [ -z "$_profile_envvar_val" ]
    then
        ! $CRON && [ -n "$ALLFUNCS_PROFILE" ] && echo "$(date): Loading $_profile_env"
        # shellcheck disable=SC1090
        . "$PERSONALBIN/$_profile_env"
    else
        ! $CRON && [ -n "$ALLFUNCS_PROFILE" ] && echo "$(date): Skipped loading $_profile_env, $_profile_envvar=$_profile_envvar_val"
    fi
done
```

It was all easily reverted by doing a `git stash` on my latest changes to my .profile (you **do** use `git` on your dotfiles, right?), logging in again, 'git stash pop` and `git diff`. But I think shellcheck should have flagged that in my editor Friday when I changed it.
